src/class-validator.php

```diff
@@ -68,4 +79,51 @@ public static function validate_api_secret( string $api_secret ): bool {
 	public static function validate_metadata_secret( string $metadata_secret ): bool {
 		return strlen( $metadata_secret ) === 10;
 	}
+
+	/**
+	 * Validates the passed API Credentials.
+	 *
+	 * @since 3.11.0
+	 *
+	 * @param Parsely $parsely The Parsely instance.
+	 * @param string  $site_id The Site ID to be validated.
+	 * @param string  $api_secret The API Secret to be validated.
+	 * @return WP_Error|true True if the API Credentials are valid, false otherwise.
+	 */
+	public static function validate_api_credentials( Parsely $parsely, string $site_id, string $api_secret ) {
+		$wp_error = new WP_Error();
+
+		if ( ! self::validate_site_id( $site_id ) ) {
+			$wp_error->add( self::INVALID_SITE_ID, __( 'Invalid Site ID', 'wp-parsely' ) );
+		}
+
+		if ( ! self::validate_api_secret( $api_secret ) ) {
+			$wp_error->add( self::INVALID_API_SECRET, __( 'Invalid API Secret', 'wp-parsely' ) );
+		}
+
+		if ( $wp_error->has_errors() ) {
+			return $wp_error;
+		}
```

Comment:
It is just validating the form (length and format) of the Site ID and API secret. It might save a few unnecessary API requests with invalid values. But I do agree it might be redundant and not offer any tactile gain in having both, apart from an API request here and there. 

What do you think? Is it worth to keep it, or rely exclusively on the API validation?

Reply:
I'd go for API exclusively, since the validations in place aren't that good anyway and were just a compromise until we had proper validation. I totally agree that having them could save us some API trips, but the Settings page is a set-and-forget thing that most customers don't come back to. So I think removing them is fine in this case.